sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -384,6 +384,17 @@ export class EventHubSender extends LinkEntity {
     options?: SendOptions & EventHubProducerOptions
   ): Promise<void> {
     try {
+      if (events instanceof EventDataBatch && options && options.partitionKey) {
+        // throw an error if partition key is different than the one provided in the options.
+        const error = new Error("Partition key is not supported when sending a batch message. Pass the partition key when creating the batch message instead.");
+        log.error(
+          "[%s] Partition key is not supported when sending a batch message. Pass the partition key when creating the batch message instead. %O",
+          this._context.connectionId,
+          error
+        );
+        throw error;
+      }
```

Comment:
What was the reason for changing the order of checks here?
I believe previously we had the check for partition Key  + partitionId first

Reply:
In case of `EventDataBatch`  if we first check `options && options.partitionKey` and throw error `Partition key is not supported when sending a batch message. Pass the partition key when creating the batch message instead` then it will not check ` typeof options.partitionKey === "string" && typeof options.partitionId === "string"` case and I think this is what we want.

If we write ` typeof options.partitionKey === "string" && typeof options.partitionId === "string"` check first then it will throw `Partition key is not supported when using producers that were created using a partition id.`

So I thought if user will pass the partitionKey then we will give user the batch error and it will not check second check.